base/string.jl

```diff
@@ -1,11 +1,21 @@
 ## core text I/O ##
 
 print(io::IO, x) = show(io, x)
-function print(io::IO, xs...)
-    lock(io) do io
-        for x in xs print(io, x) end
+stagedfunction print(io::IO, xs...)
+    if :lock in fieldnames(io)
```

Comment:
I'd rather this not be a staged function. It's overkill, since it will force recompilation for every combination of trailing arguments, whose types aren't even used. Is an `isdefined` check too slow?